diagnostic.md

```diff
@@ -2,53 +2,66 @@
 
 Place your responses inside the fenced code-blocks where indicated by comments.
 
-
 What is the purpose of a backend?
 
-```bash
-// your response here
+```md
+To store and manipulate data for your application.
 ```
 
 Which layer in the MVC pattern is used by the controller to fetch data?
 
-```bash
-// your response here
+```md
+model
 ```
 
 In which part of the MVC pattern can we find C.R.U.D actions?
 
-```bash
-// your response here
+```md
+controller
 ```
+
 List at least 5 standard actions that C.R.U.D corresponds to?
 
-```bash
-// your response here
+```md
+1. index
+2. show
+3. create
+4. update
+5. destroy
 ```
 
 A user action fires a `PATCH` request for `pets/1`. Explain in detail each step
 required for data to be returned to the client. (bullet points or ordered list,
 please include information on dynamic segments, the params hash and seralizers).
 
-```bash
-// your response here
+```md
+1. The router receives the PATCH request for 'pets/1' and passes it on to the
+   correct C.R.U.D. action in the controller based on the dynamic segments.
+   C.R.U.D. actions are defined in the params hash in the controller.
```

Comment:
the crud actions are defined in the controller.  The params hash is constructed with the information you passed to the route.